streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -461,121 +466,155 @@ public Set<String> makeReady(final Map<String, InternalTopicConfig> topics) {
         // have existed with the expected number of partitions, or some create topic returns fatal errors.
         log.debug("Starting to validate internal topics {} in partition assignor.", topics);
 
-        long currentWallClockMs = time.milliseconds();
+        final long currentWallClockMs = time.milliseconds();
         final long deadlineMs = currentWallClockMs + retryTimeoutMs;
 
-        Set<String> topicsNotReady = new HashSet<>(topics.keySet());
+        final Set<String> topicsNotReady = new HashSet<>(topics.keySet());
         final Set<String> newlyCreatedTopics = new HashSet<>();
 
         while (!topicsNotReady.isEmpty()) {
             final Set<String> tempUnknownTopics = new HashSet<>();
-            topicsNotReady = validateTopics(topicsNotReady, topics, tempUnknownTopics);
-            newlyCreatedTopics.addAll(topicsNotReady);
 
+            // Only consider the not-ready subset on each iteration
+            final Map<String, InternalTopicConfig> notReadyTopicsMap = topics.entrySet().stream()
+                    .filter(e -> topicsNotReady.contains(e.getKey()))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            final Set<NewTopic> topicsToCreate = computeTopicsToCreate(notReadyTopicsMap, tempUnknownTopics);
```

Comment:
Here we compute `topicToCreate`, but for all existing topics, we don't return them, but we also don't remove existing topics from `topicsNotReady`.

I think, we need to add a step `topicsNotReady.retainAll(notReadyTopicsMap.keySet())` after this line, plus modify `notReadyTopicsMap` inside `computeTopicsToCreate` and remove all entries for the topics we successfully verify.